### PR TITLE
API endpoint update, refactored file creation and compression

### PR DESCRIPTION
**Pull request by braddeibert**
CHANGES:
- Added GET request handling to existing API endpoint
- Added new 'run' parameter to requests. Allows users to choose between receiving the FDS run output or only the FDS input file
- Refactored file creation and compression process in *landfireCurl.py*. Files created during backend process are now removed after they've been compressed into output .zip file.
- 5MB size hard limit now being enforced on data directory. *data/* is cleared if size exceeds this threshold
- Edits to HTML form and README.md to reflect updates to API

TESTING:
- Test both POST and GET requests against the endpoint. (using webpage form for POST requests and curl for GET requests)
- Be sure that new 'run' option works. (FDS runs are only triggered when run="true", and only the FDS input file is returned when run="false")
- Check that compressed files contain expected output

### CODE REVIEW

File: landfireCurl.py [hunk 1]
```diff
@@ -2,41 +2,48 @@
...
         # builds unique(ish) folder name system
         today = date.today()
         d2 = today.strftime("%Y-%m-%d-")
         self.title = d2+saveName.split('.')[0]
 
-        if not os.path.isdir('data'):
-            os.mkdir('data')
-        os.chdir('data')
+        if os.getcwd().split('/')[-1] != 'data':
+            os.chdir('data')
```
Comment (tjschweitzer) [inline landfireCurl.py:31]: With the data directory being part of the git ignore the directory does not exist the first time this repository is ran after being pulled. Need to check if data is a directory if not then mkdir and then chdir

File: landfireCurl.py [hunk 2]
```diff
@@ -2,41 +2,48 @@
...
 
-        if not os.path.isdir(self.title):
-            os.mkdir (self.title)
-        else:
-            print("Directory already exists some files may be over written")
-        os.chdir(self.title)
+        # clear data directory if size exceeds 5MB (5,000,000 bytes)
+        if _dirSize(os.getcwd()) > 5000000:
+            os.system("rm -rf *")
```
Comment (tjschweitzer) [inline landfireCurl.py:35]: This command will cause no issues as long as only one simulation is running in the container at a time. If an fds run is going and another request is started wouldn't we lose the data from the fds run in progress?

**Review by tjschweitzer — CHANGES_REQUESTED** [review]

File: landfireCurl.py (hunk 2 above)
Comment (braddeibert) [reply landfireCurl.py:35]: Didn't change this for now, figure we can iterate back later if it becomes a problem.

File: landfireCurl.py (hunk 1 above)
Comment (braddeibert) [reply landfireCurl.py:31]: Changed .gitignore to track data/ but none of its contents.

**Comment by braddeibert** [comment]
@tjschweitzer changes are added. The README now has examples for using the endpoint and the data/ issue is resolved.

**Pull request merged**